### PR TITLE
[codex] Fix wide glyph output in legacy Windows console

### PR DESCRIPTION
**Pull request by BruceChenQAQ**
Related: #602

## Summary
- narrow emoji variation selector injection so only real emoji and wide symbol glyphs get `U+FE0F`
- skip zero-width placeholder cells in `DefaultNetConsoleOutput` so the Windows legacy console path does not write `NUL` bytes for wide glyph tails
- add regression coverage for full-width punctuation and legacy-console wide-glyph output

## Root Cause
`Symbol(char)` previously treated every width-2 non-letter as an emoji-style glyph and appended `U+FE0F`. That was too broad: full-width punctuation such as `：` `。` `（` `）` `！` is naturally double-width in CJK terminals and must stay as plain text.

A second issue remained in the legacy Windows console path. Wide glyph rendering uses zero-width placeholder cells for the trailing columns of a double-width glyph, but `DefaultNetConsoleOutput.WritePixel()` still wrote those placeholder cells. On the .NET console API this serialized a `\0` character into the output stream, which corrupted subsequent text layout in `WindowsLegacyConsoleOutput`.

## Fix
- restrict emoji-style variation injection to actual emoji and `UnicodeCategory.OtherSymbol` wide glyphs
- return early for `pixel.Width <= 0` in `DefaultNetConsoleOutput.WritePixel()`, matching the ANSI writer's behavior
- add tests covering both the punctuation case and the zero-width placeholder case

## Validation
- `dotnet test src/Tests/Consolonia.Core.Tests/Consolonia.Core.Tests.csproj -c Release --filter "FullyQualifiedName~DefaultNetConsoleOutputTests|FullyQualifiedName~SimpleSymbolTests|FullyQualifiedName~GlyphTests"`
- `dotnet build src/Consolonia/Consolonia.csproj -c Release`
- downstream manual validation in Minecraft Console Client using a Windows self-contained single-file publish launched by double-click, which forces the legacy console output path

### CODE REVIEW

**Comment by jinek** [comment]
@BruceChenQAQ Thank you. Could you please confirm whether you can click the failing build to see the error?

**Comment by BruceChenQAQ** [comment]
@jinek Yes. I checked the failing build and the error was in `Consolonia.Gallery.Tests`: `FontsTests.DisplaysWideTermText` was still asserting the old incorrect WideTerm output with injected `U+FE0F` selectors.

The rendering fix itself was already correct. I updated the outdated test expectation in `5c531e6` and reran `dotnet test src/Consolonia.sln` locally:
- `Consolonia.Core.Tests`: 270 passed, 1 skipped
- `Consolonia.Gallery.Tests`: 40 passed, 2 skipped

That commit retriggered the PR checks.

File: src/Consolonia.Core/Infrastructure/DefaultNetConsoleOutput.cs [hunk 1]
```diff
@@ -59,6 +59,9 @@ public virtual PixelBufferCoordinate GetCaretPosition()
 
         public virtual void WritePixel(PixelBufferCoordinate position, in Pixel pixel)
         {
+            if (pixel.Width <= 0)
```
Comment (jinek) [inline src/Consolonia.Core/Infrastructure/DefaultNetConsoleOutput.cs]: What is the case of negative width symbol? Im confused here because we are skipping the rest of the code, which does flushing and other stuff.

Comment (jinek) [reply src/Consolonia.Core/Infrastructure/DefaultNetConsoleOutput.cs]: @coderabbitai any comments here?

Comment (jinek) [reply src/Consolonia.Core/Infrastructure/DefaultNetConsoleOutput.cs]: But further when writing to output, when flushing, isn't it expected to have 0 width cells?

Comment (BruceChenQAQ) [reply src/Consolonia.Core/Infrastructure/DefaultNetConsoleOutput.cs]: Negative width can't really happen here. `pixel.Width` on this path is effectively only used as 0 / 1 / 2, so this guard is really about the `width == 0` case.

That zero-width case is the trailing placeholder cell of a wide glyph. On the legacy `Console` API path we want to skip it completely, otherwise we end up writing the placeholder char itself and that is what corrupts the layout.

Skipping it also does not miss any required flush, because there is nothing to emit for that cell, and the frame still ends with the normal final `_console.Flush()` in `RenderTarget`.

I pushed a small follow-up to make that clearer: changed the condition from `<= 0` to `== 0` and added a comment explaining the placeholder case.

**Review by jinek — APPROVED** [review]

**Comment by jinek** [comment]
@coderabbitai why isnt this pr merged automatically?

**Comment by jinek** [comment]
> 
> > [!TIP]
> > For best results, initiate chat on the files or code changes.
> 
> `@jinek` CodeRabbit does not merge pull requests — it is an AI code reviewer. Merging is a human action. To merge this PR, you (as a maintainer) can click **Merge pull request** on GitHub directly.
> 
> If you'd like me to approve it first, you can comment `@coderabbitai approve` on the PR.

Ive clickef automatic merge, but something prevents it.

@coderabbitai
